src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -346,28 +380,37 @@ export class DataSourceSavedObjectsClientWrapper {
             'Update failed due to deprecated data source: "auth.credentials.username" missing. Please delete and create another data source.'
           );
         }
+        return this.validateEncryptionContext(password, endpoint);
```

Comment:
why do we need to validate encryption context in case of getting encryption context?